pkg/apis/sessions/session_state.go

```diff
@@ -64,6 +64,32 @@ func (s *SessionState) String() string {
 	return o + "}"
 }
 
+func (s *SessionState) GetClaim(claim string) string {
+	if s == nil {
+		return ""
+	}
+	switch claim {
+	case "access_token":
+		return s.AccessToken
+	case "id_token":
+		return s.IDToken
+	case "created_at":
+		return s.CreatedAt.String()
+	case "expires_on":
+		return s.ExpiresOn.String()
+	case "refresh_token":
+		return s.RefreshToken
+	case "email":
+		return s.Email
+	case "user":
+		return s.User
+	case "preferred_username":
+		return s.PreferredUsername
```

Comment:
Once we allow generic claims, I'd really like to see this one be an optional claim users can configure to be added to the session & we leave our 3 cardinal claims that are always part of the session as 'User, Email & Groups'

Reply:
100% agree, let's preserve the behaviour for now but remove in a future release

Reply:
Any ETA for generic claims name ? Thanks for your works

Reply:
We'll begin work on it and accept community PRs for it now. V7 which just released was Authentication/Authorization focused.

V8 Will have a major focus on Structured Configs -- and with this rich configuration format, a much easier mechanism to specify adhoc claim -> session -> header mappings.

The legacy/current style flag based flat configs didn't lend themselves well to this implementation.